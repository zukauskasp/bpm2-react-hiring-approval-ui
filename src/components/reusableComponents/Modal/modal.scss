.modal-container {
    min-width: 500px;

    .modal-buttons-container{
        display: flex;
        width: 100%;
        padding-bottom: 40px;
        justify-content: center;
    }
    .modal-content-container{
        padding: 0 73px 30px 73px;
        
        p {
            text-align: center;
            font-size: 16px;
            line-height: 1.5;
            color: #003755;
            max-width: 306px;

        }
    }
    
    .close-container {
        width: 100%;
        .close-icon {
            float: right;
            width: 24px;
            height: 24px;
            margin: 8px;
            cursor: pointer;
        }
    }
    
}
