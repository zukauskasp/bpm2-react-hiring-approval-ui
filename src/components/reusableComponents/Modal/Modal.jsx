import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import DialogTitle from '@material-ui/core/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent';
import Dialog from '@material-ui/core/Dialog';
import CustomButton from '../Button/Button';
import Icon from "../Icon/Icon";

import './modal.scss';

const Modal = ({children,title,onClose,open,submitButtonText,onSubmit } ) => {
    const classes = useStyles();

	return (
		<Dialog onClose={onClose} open={open} maxWidth={false}>
			<div className="modal-container">
				<div className='close-container'>
					<Icon icon="remove" onClick={onClose} className='close-icon' />
				</div>
				<DialogTitle disableTypography={true} className={classes.title}>{title}</DialogTitle>
				<DialogContent className={classes.content}>

					<div className="modal-content-container">
						{children}
					</div>

					<div className="modal-buttons-container">
						<CustomButton
							text="Cancel"
							buttonType="secondary"
							onClickEvent={onClose}
						/>
						<CustomButton
							text={submitButtonText}
							buttonType="primary"
							onClickEvent={onSubmit}
						/>
					</div>
				</DialogContent>
			</div>
		</Dialog>
    )
}

const useStyles = makeStyles({
	title: {
		textAlign: 'center',
		fontSize: '36px',
		fontWeight: '500',
		paddingTop: '40px;',
		fontFamily: "Danske Human Medium Italic, Danske Human",
		color: '#003755'
	},
	content: {
		paddingTop: '20px',
	}
});

Modal.propTypes = {
	onClose: PropTypes.func.isRequired,
	submitButtonText: PropTypes.string,
	open: PropTypes.bool,
	title: PropTypes.string,
	onSubmit: PropTypes.func.isRequired,
}

Modal.defaultProps = {
	submitButtonText: 'Submit',
	open: false,
}

export default Modal;