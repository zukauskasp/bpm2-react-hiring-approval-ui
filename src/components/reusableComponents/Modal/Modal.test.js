import React from 'react';
import {shallow,mount} from 'enzyme';
import Modal from './Modal';

describe('Modal', ()=> {
    
    afterAll(() => {
        mount.cleanUp();
      });

    it('expect to match latest snapshot', () => {
        const wrapper =  mount(<Modal onSubmit={()=>{}} title="title" onClose={()=>{}}>Children text</Modal>);
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it('expect to render modal component', () => {
        expect(shallow(<Modal onSubmit={()=>{}} onClose={()=>{}}/>).length).toEqual(1);
    }); 

    it('expect to have children rendered', () => {
        const wrapper = shallow(<Modal onSubmit={()=>{}} onClose={()=>{}}>Children text</Modal>);
        expect(wrapper.find('.modal-content-container').contains('Children text')).toEqual(true);
    });

    it('expect to have close icon/img rendered', () => {
        const wrapper = shallow(<Modal onSubmit={()=>{}} onClose={()=>{}}>Children text</Modal>);
        expect(wrapper.find('.close-icon').length).toEqual(1);
    });

});