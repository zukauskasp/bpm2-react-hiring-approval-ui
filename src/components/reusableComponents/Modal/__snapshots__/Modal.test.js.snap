// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal expect to match latest snapshot 1`] = `
"<Modal onSubmit={[Function: onSubmit]} title=\\"title\\" onClose={[Function: onClose]} submitButtonText=\\"Submit\\" open={false}>
  <WithStyles(ForwardRef(Dialog)) onClose={[Function: onClose]} open={false} maxWidth={false}>
    <ForwardRef(Dialog) classes={{...}} onClose={[Function: onClose]} open={false} maxWidth={false}>
      <ForwardRef(Modal) className=\\"MuiDialog-root\\" BackdropComponent={{...}} BackdropProps={{...}} closeAfterTransition={true} disableBackdropClick={false} disableEscapeKeyDown={false} onEscapeKeyDown={[undefined]} onClose={[Function: onClose]} open={false} />
    </ForwardRef(Dialog)>
  </WithStyles(ForwardRef(Dialog))>
</Modal>"
`;
